pkg/api/defaults-apiserver.go

```diff
@@ -21,10 +21,9 @@ func (cs *ContainerService) setAPIServerConfig() {
 		"--insecure-port":               "8080",
 		"--secure-port":                 "443",
 		"--service-account-lookup":      "true",
-		"--etcd-cafile":                 "/etc/kubernetes/certs/ca.crt",
 		"--etcd-certfile":               "/etc/kubernetes/certs/etcdclient.crt",
 		"--etcd-keyfile":                "/etc/kubernetes/certs/etcdclient.key",
-		"--etcd-servers":                "https://127.0.0.1:" + strconv.Itoa(DefaultMasterEtcdClientPort),
+		"--etcd-servers":                "https://<etcdEndPointUri>:" + strconv.Itoa(DefaultMasterEtcdClientPort),
```

Comment:
Can we make the substitution in-code instead of templating and seding? I believe it should be possible within this method to do that.

Reason is that we have recently been moving towards removing sed statements and keeping custom scripts to a smaller size :). This is also more testable IMO

Reply:
@tariq1890 good callout: https://github.com/Azure/aks-engine/issues/100